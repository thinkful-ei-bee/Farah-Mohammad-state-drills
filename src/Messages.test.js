import React from 'react';
import ReactDOM from 'react-dom';
import Messages from './Messages';
import renderer from 'react-test-renderer';

// smoke test
describe ('My component', () => {
it ('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Messages />, div);
    ReactDOM.unmountComponentAtNode(div);
});

// snapshot test 
it('renders the UI as expected', () => {
    const tree = renderer
      .create(<Messages name="Messages" unread={4}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
});

it('renders the UI as expected with no unreads', () => {
    const tree = renderer
      .create(<Messages name="Messages" unread={0}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
});
});