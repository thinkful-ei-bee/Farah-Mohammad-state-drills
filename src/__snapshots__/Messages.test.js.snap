// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My component renders the UI as expected 1`] = `
<div
  className="messages"
>
  Messages
  <div
    className="unread_count badge"
  >
    4
  </div>
</div>
`;

exports[`My component renders the UI as expected with no unreads 1`] = `
<div
  className="messages"
>
  Messages
</div>
`;
